wasp/src/main/java/com/orhanobut/wasp/utils/MockFactory.java

```diff
@@ -0,0 +1,209 @@
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Factory util for creating populated mock response objects.
+ *
+ * @author Oguz Babaoglu
+ */
+public final class MockFactory {
+
+    private static final int MOCK_LIST_COUNT = 5;
+
+    private MockFactory() {
+        //no instance
+    }
+
+    /**
+     * Reads mock response string from given path.
+     *
+     * @param context  Context with file access
+     * @param filePath Path to mock file
+     * @return Response string
+     */
+    public static String readMockResponse(Context context, String filePath) {
+
+        String responseString;
+        try {
+            responseString = IOUtils.readFileFromAssets(context, filePath);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        if (TextUtils.isEmpty(responseString)) {
+            throw new RuntimeException("Mock file \"" + filePath + "\" is empty");
+        }
+
+        return responseString;
+    }
+
+    /**
+     * Creates a mock object and populates its fields.
+     *
+     * @param type Type of mock object
+     * @param <T>  Type of mock object
+     * @return Mock Object of type T
+     */
+    @SuppressWarnings("unchecked") // types must agree
+    public static <T> T createMockObject(Type type) {
+        Class<T> rawType;
+
+        if (type instanceof ParameterizedType) {
+            rawType = (Class<T>) ((ParameterizedType) type).getRawType();
+        } else if (type instanceof Class) {
+            rawType = (Class<T>) type;
+        } else {
+            throw new UnsupportedOperationException("Unsupported type: "
+                    + type.getClass().getSimpleName());
+        }
+        T instance = instantiateObject(rawType);
+
+        return populateObject(instance);
+    }
+
+    /**
+     * Creates a new instance of given raw type.
+     *
+     * @param rawType Raw Type of object
+     * @param <T>     Type of object
+     * @return Object instance
+     */
+    @SuppressWarnings("unchecked") // types must agree
+    public static <T> T instantiateObject(Class<T> rawType) {
+        try {
+            Constructor<?> constructor = rawType.getDeclaredConstructor();
+            if (!constructor.isAccessible()) {
+                constructor.setAccessible(true);
+            }
+            return (T) constructor.newInstance((Object[]) null);
+        } catch (Exception e) {
+            // Default constructor failed, attempt Unsafe Allocation
+        }
+
+        try {
+            UnsafeAllocator unsafeAllocator = UnsafeAllocator.create();
+            return unsafeAllocator.newInstance(rawType);
+        } catch (Exception e) {
+            // Give up
+            throw new RuntimeException("Failed to instantiate "
+                    + rawType.getSimpleName(), e);
```

Comment:
instead of simple name, I think canonical would suit better. There might be the same class name in different packages.